extensions/wikia/SpecialPageViews/SpecialPageViewsController.class.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+class SpecialPageViewsController extends WikiaSpecialPageController {
+
+	const SPECIALPAGE_NAME = 'PageViews';
+
+	private $startDate, $endDate;
+
+	function __construct() {
+		parent::__construct( self::SPECIALPAGE_NAME );
+		$this->wg->Out->addModules( 'ext.SpecialPageViews' );
+	}
+
+	public function execute() {
+		wfProfileIn( __METHOD__ );
+		$this->wg->Out->setPageTitle( wfMessage( 'admindashboard-title' )->escaped() );
+
+		if ( !$this->wg->User->isAllowed( 'pageviews' ) ) {
+			$this->displayRestrictionError();
+			return false;  // skip rendering
+		}
+
+		$this->setHeaders();
+		$this->setInitialDates();
+
+		$oOutput = ( new SpecialPageViewsOutput );
+		$oOutput->set( $this->getReport() );
+		$this->setJsVars( $oOutput );
+		$sReturnChart = $oOutput->getHTML();
+		$this->wg->Out->addHTML( $sReturnChart );
+
+		wfProfileOut( __METHOD__ );
+	}
+
+	public function getData() {
+		$inStartDate = $this->getVal( 'startDate', date( 'Y-m-d' ) );
+		$inEndDate = $this->getVal( 'endDate', date( 'Y-m-d' ) );
+		$this->startDate = DateTime::createFromFormat( 'Y-m-d', $inStartDate );
+		$this->endDate = DateTime::createFromFormat( 'Y-m-d', $inEndDate );
+
+		unset( $inStartDate, $inEndDate );
+
+		if ( !$this->startDate || !$this->endDate ) {
+			$this->setInitialDates();
+		}
+
+		$oOutput = ( new SpecialPageViewsOutput );
+		$oOutput->set( $this->getReport( $this->startDate, $this->endDate ) );
+
+		$mOut = $oOutput->getRaw();
+		$this->setVal('chartId', $mOut->chartId );
+		$this->setVal('datasets', $mOut->datasets );
+		$this->setVal('fullTicks', $mOut->fullTicks );
+		$this->setVal('hiddenSeries', $mOut->hiddenSeries );
+		$this->setVal('monthly', $mOut->monthly );
+		$this->setVal('ticks', $mOut->ticks );
+	}
+
+	public function setHeaders() {
+		$oOut = $this->wg->Out;
+		$oOut->setRobotPolicy( 'noindex,nofollow' );
+		$oOut->setPageTitle( wfMessage( 'special-pageviews-title' )->escaped() );
+	}
+
+	private function getReport() {
+		$oReport = ( new SponsorshipDashboardReport );
+		$oReport->name = wfMsg( 'special-pageviews-report-title' );
```

Comment:
And from a special page, use `$this->msg( 'message-key' )->escaped()` instead of `wfMessage`.